### PR TITLE
Add type hints for Mood class

### PR DESCRIPTION
**Pull request by ggravlingen**

### CODE REVIEW

File: pytradfri/mood.py [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
         """Create object of class."""
         super().__init__(raw)
         self._parent = parent
 
     @property
-    def path(self):
+    def path(self) -> list[str]:
         """Path."""
-        return [ROOT_MOODS, self._parent, self.id]
+        return [ROOT_MOODS, self._parent, cast(str, self.id)]
```
Comment (MartinHjelmare) [inline pytradfri/mood.py:21]: The `id` is typed as an integer. So we need to copy to string if we want a string. We can only use `cast` when we want to tell mypy what type something is. It doesn't change the type.
```suggestion
        return [ROOT_MOODS, self._parent, str(self.id)]
```

Comment (ggravlingen) [reply pytradfri/mood.py:21]: OK, thanks!

**Review by MartinHjelmare — APPROVED** [review]

**Comment by ggravlingen** [comment]
I'll fix the test

**Comment by ggravlingen** [comment]
I fixed it but should we really have to convert to str in the path property? We should have receved a string from parent class?

**Comment by ggravlingen** [comment]
Will not merge until you have commented on the last change.

File: pytradfri/mood.py [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
         """Create object of class."""
         super().__init__(raw)
         self._parent = parent
 
     @property
-    def path(self):
+    def path(self) -> list[str]:
         """Path."""
-        return [ROOT_MOODS, self._parent, self.id]
+        return [ROOT_MOODS, str(self._parent), str(self.id)]
```
Comment (MartinHjelmare) [inline pytradfri/mood.py:19]: Yeah, the parent should be a string already according to our typing. We need to check that if we think it's not true.

Comment (ggravlingen) [reply pytradfri/mood.py:19]: Odd, it looks like strings here:
https://github.com/home-assistant-libs/pytradfri/blob/2fc8fa0ec1a639a0ace6093b1311d2009d453123/pytradfri/gateway.py#L212

Comment (ggravlingen) [reply pytradfri/mood.py:19]: Oh, I believe moods linked to a group is of type `int` and not `str`. Let me push that code and we can have a look at both at the same time.

Comment (MartinHjelmare) [reply pytradfri/mood.py:19]: We didn't resolve this, I think.

Comment (ggravlingen) [reply pytradfri/mood.py:19]: Sorry, I misunderstood you in the other PR. This is what I meant by converting to str and the looking into it later. Should we revert PR?

**Pull request merged**